extensions/sd_api_pictures/script.py

```diff
@@ -111,31 +129,57 @@ def input_modifier(string):
 
     return string
 
+def apply_settings():
+
+    payload_options = {
+        "sd_model_checkpoint": params['sd_model_checkpoint']
+        #"sd_vae": params['sd_vae']
+    }
+
+
+    print(f'Updating parameters via the API on {params["address"]}...')
+    response = requests.post(url=f'{params["address"]}/sdapi/v1/options', json=payload_options)
+    response.raise_for_status()
+    r = response.json()    
+
+
 # Get and save the Stable Diffusion-generated picture
 def get_SD_pictures(description):
-
+   
     global params
 
     if params['manage_VRAM']:
         give_VRAM_priority('SD')
 
+
+
     payload = {
+
         "prompt": params['prompt_prefix'] + description,
+        "negative_prompt": params['negative_prompt'],
         "seed": params['seed'],
-        "sampler_name": params['sampler_name'],
-        "steps": params['steps'],
+        "sampler_index": params['sampler_index'],
```

Comment:
This change is incorrect, `index` is a very old legacy key-name. `sampler_name` was right.

Reply:
I was trying to get the samplers to change with name but somehow, they only picked up with index but if name works leave name.